docs/strategy-callbacks.md

```diff
@@ -783,7 +783,7 @@ Additional entries are ignored once you have reached the maximum amount of extra
 ### Decrease position
 
 The strategy is expected to return a negative stake_amount (in stake currency) for a partial exit.
-Returning the full owned stake at that point (based on the current price) (`-(trade.amount / trade.leverage) * current_exit_rate`) results in a full exit.  
```

Comment:
> Returning -trade.stake_amount will already trigger the full exit

Unfortunately, that's incorrect (or rather, is only correct if current_rate == entry-rate).

The check happens on `remaining` - which is calculated as `(trade.amount - amount) * current_exit_rate`.
This won't be `trade.stake_amount` necessarily - and definitely isn't if price moved ever so slightly.

https://github.com/freqtrade/freqtrade/blob/65a09b0d89f5ec313713215fafafbec1c423bf66/freqtrade/freqtradebot.py#L676-L677

I tried, but couldn't find a check which won't have accidental side-effects to support returning stake_amount directly.

We could check for `result == trade.stake_amount` - but there's too much possibility for odd edge-cases - especially once you move into the "higher win" territory - e.g. a position that opened for 100$ - has 150% profit (so it's valued at 250$). Now the user tries to sell 100$ of that as partial sell - so he's returning `-100` (take out the initial stake, let profits run further)  ... the check for "trade.stake_amount == response" would trigger - though the resulting full exit is VERY far away from what the user intended.

You can play with this yourself - simply changing the following test-case to return `-trade.stake_amount` - and you'll see what i mean (the test doesn't use 150% profit - but still shows that -stake_amount != what the docs said before).

https://github.com/freqtrade/freqtrade/blob/65a09b0d89f5ec313713215fafafbec1c423bf66/tests/freqtradebot/test_integration.py#L669-L671

Personally, i think if you need full exits - the simple solution is to go with `custom_exit()`.
However this change is wrong - unless you can also change the linked test above to correspond (and not fail) ... however without side-effects - which is the problem i encountered.

Reply:
min_exit_stake is a value based on stake? Why are you comparing it to amount (that is in coin value)? I thought remaining supposed to be trade.stake_amount - stake (which is returned from the adjust function?

Reply:
I'll try to play with it

Reply:
it's not - and never really was (at least not as far as i remember this part of the code - though a git blame might proof me wrong - as it's purely based on memory - not on checking😆 )